src/constructs/s3/index.ts

```diff
@@ -19,3 +22,130 @@ export class GuS3Bucket extends GuAppAwareConstruct(Bucket) {
     });
   }
 }
+
+interface GuS3OriginBucketDeploymentProps {
+  /**
+   * Customises the SSM Parameter path, making it stage agnostic.
+   * Useful when a single bucket is used across multiple stages.
+   *
+   * When `false` the path will be `/<STAGE>/<STACK>/<APP>/<APP>-origin-bucket`.
+   * When `true` the path will be `/<STACK>/<APP>/<APP>-origin-bucket`.
+   *
+   * @default false
+   */
+  withoutStageAwareness: boolean;
```

Comment:
Also the [recommendation](https://github.com/guardian/recommendations/discussions/74) is to have a bucket per stage, the `aws-s3` deployment type has a `prefixStack` parameter, defined as:

```
Prefix the S3 bucket key with the target stack. Default true.
```

That is, I suspect historic usage of `aws-s3` would indicate that we have a shared bucket.